homeassistant/components/prometheus/__init__.py

```diff
@@ -535,90 +565,75 @@ def _handle_climate(self, state: State) -> None:
         )
 
         if current_action := state.attributes.get(ATTR_HVAC_ACTION):
-            metric = self._metric(
-                "climate_action",
-                prometheus_client.Gauge,
-                "HVAC action",
-                ["action"],
-            )
             for action in HVACAction:
-                metric.labels(**dict(self._labels(state), action=action.value)).set(
-                    float(action == current_action)
-                )
+                self._metric(
+                    "climate_action",
+                    prometheus_client.Gauge,
+                    "HVAC action",
+                    self._labels(state) | {"action": action.value},
```

Comment:
Not strictly related to your change, but I wonder if we should have `self._labels` accept an optional `extra_labels` argument, and make it check for duplicate keys before merging the dicts?

Reply:
Done.